node/src/components/consensus/era_supervisor.rs

```diff
@@ -645,6 +653,12 @@ where
                     self.era_supervisor.active_eras[&era_id].start_height + height,
                     proposer,
                 );
+                let time_since_proto_block = finalized_block.timestamp().elapsed().millis();
+                self.era_supervisor
+                    .metrics
+                    .finalization_time
+                    .set(time_since_proto_block as f64);
+                self.era_supervisor.metrics.amount_of_blocks.inc();
```

Comment:
To clarify, this metric will answer the question:
"How much time passed between the _proposal_ and the _finalization_ of the most recently finalized block."
It does not answer:
"When was the last time a new block was proposed."
(Not sure what's the intent; we should just document it correctly.)

Reply:
> "When was the last time a new block was proposed."

Maybe to disambiguate even further: _block_ to me sounds like at best the leader message. We should track when was the last time validator created _any_ consensus message. Whether that message is a leader block or not depends on the seed and leader schedule. I am not sure if that's an interesting metric (we don't want to debug leader schedule based on metrics).

Reply:
> last time validator created any consensus message

Maybe, but that would be a completely different metric, and wouldn't make this one redundant, would it?

Reply:
_this one_ meaning the one that Karan is proposing? I don't think so. We can be adding consensus messages to the protocol state and still not finalize leader block.

Reply:
> "When was the last time a new block was proposed."

This doesn't sound like a metric. It should go to `/status` endpoint IMO, and we should have filter on it in the log viewer but it's not suitable for metrics.